xcode/Subconscious/Shared/Components/Common/Byline/PetnameView.swift

```diff
@@ -22,13 +21,13 @@ extension UserProfile {
             self.address.petname?.leaf
         ) {
         case (.ourself, _, .some(let selfNickname), _):
-            return NameVariant.petname(Slashlink.ourProfile, selfNickname)
+            return NameVariant.known(Slashlink.ourProfile, selfNickname)
         case (_, .following(let petname), _, _):
-            return NameVariant.petname(self.address, petname)
+            return NameVariant.known(self.address, petname)
         case (_, .notFollowing, .some(let selfNickname), _):
-            return NameVariant.selfNickname(self.address, selfNickname)
+            return NameVariant.unknown(self.address, selfNickname)
         case (_, .notFollowing, _, .some(let proposedName)):
-            return NameVariant.proposedName(self.address, proposedName)
+            return NameVariant.unknown(self.address, proposedName)
```

Comment:
Simplified the modelling for name display, we either know a user or we don't. 

If we know them, we use our name for them. If we don't know them, we use their nickname (if available) or fall back to the leaf of their current address.

Reply:
I like the clarity, however, I want to note that this loses us information.

It seems likely at some point we will want to be able to distinguish proposed names from self-proposed names (https://github.com/cwebber/rebooting-the-web-of-trust-spring2018/blob/petnames/draft-documents/petnames.md). For example,

- We may want to prefer a self-proposed name to a 3p-proposed name when following and suggesting a name.
- (Speculative) We may wish, in future, to show the user's self-proposed name in context menus or place a badge next to self-proposed names to show user's preferred name.

Reply:
It doesn't actually lose us any information in the model itself, just in the mapping to an (unused) view construct. We were rendering `proposedName` and `selfNickname` identically before this change so nothing visibly changes as a result of this.

If the kind of cases you point out do arise, we can derive all that we need from a `UserProfile` regardless.